internal/data/wallets.go

```diff
@@ -3,45 +3,109 @@ package data
+)
+
 type Wallet struct {
-	ID                string     `json:"id" db:"id"`
-	Name              string     `json:"name" db:"name"`
-	Homepage          string     `json:"homepage,omitempty" db:"homepage"`
-	SEP10ClientDomain string     `json:"sep_10_client_domain,omitempty" db:"sep_10_client_domain"`
-	DeepLinkSchema    string     `json:"deep_link_schema,omitempty" db:"deep_link_schema"`
-	CreatedAt         *time.Time `json:"created_at,omitempty" db:"created_at"`
-	UpdatedAt         *time.Time `json:"updated_at,omitempty" db:"updated_at"`
-	DeletedAt         *time.Time `json:"-" db:"deleted_at"`
+	ID                string          `json:"id" db:"id"`
+	Name              string          `json:"name" db:"name"`
+	Homepage          string          `json:"homepage,omitempty" db:"homepage"`
+	SEP10ClientDomain string          `json:"sep_10_client_domain,omitempty" db:"sep_10_client_domain"`
+	DeepLinkSchema    string          `json:"deep_link_schema,omitempty" db:"deep_link_schema"`
+	Countries         WalletCountries `json:"countries,omitempty" db:"countries"`
+	Assets            WalletAssets    `json:"assets,omitempty" db:"assets"`
+	CreatedAt         *time.Time      `json:"created_at,omitempty" db:"created_at"`
+	UpdatedAt         *time.Time      `json:"updated_at,omitempty" db:"updated_at"`
+	DeletedAt         *time.Time      `json:"-" db:"deleted_at"`
+}
+
+type WalletInsert struct {
+	Name              string   `db:"name"`
+	Homepage          string   `db:"homepage"`
+	SEP10ClientDomain string   `db:"sep_10_client_domain"`
+	DeepLinkSchema    string   `db:"deep_link_schema"`
+	AssetsIDs         []string `db:"assets_ids"`
+	CountriesCodes    []string `db:"countries_codes"`
+}
+
+type WalletCountries []Country
+
+var _ sql.Scanner = (*WalletCountries)(nil)
+
+func (wc *WalletCountries) Scan(src interface{}) error {
+	if src == nil {
+		*wc = make(WalletCountries, 0)
+		return nil
+	}
+
+	data, ok := src.([]byte)
+	if !ok {
+		return fmt.Errorf("could not parse countries")
+	}
+	return json.Unmarshal(data, wc)
+}
+
+type WalletAssets []Asset
+
+var _ sql.Scanner = (*WalletAssets)(nil)
+
+func (wa *WalletAssets) Scan(src interface{}) error {
+	if src == nil {
+		*wa = make(WalletAssets, 0)
+		return nil
+	}
+
+	data, ok := src.([]byte)
+	if !ok {
+		return fmt.Errorf("could not parse assets")
+	}
+	return json.Unmarshal(data, wa)
 }
 
 type WalletModel struct {
 	dbConnectionPool db.DBConnectionPool
 }
 
-func (w *WalletModel) Get(ctx context.Context, id string) (*Wallet, error) {
-	var wallet Wallet
-	query := `
+const getQuery = `
 		SELECT 
-		    w.id, 
-		    w.name, 
-		    w.homepage,
-		    w.sep_10_client_domain,
-		    w.deep_link_schema,
-		    w.created_at,
-		    w.updated_at
+		    w.*, 
+			jsonb_agg(
+				DISTINCT to_jsonb(c)
+			) FILTER (WHERE c.code IS NOT NULL) AS countries,
+			jsonb_agg(
+				DISTINCT jsonb_build_object(
+					'id', a.id,
+					'code', a.code,
+					'issuer', a.issuer 
+				)
```

Comment:
Can we use `to_jsonb` in the assets part, the same way we did it for countries?

```suggestion
				DISTINCT to_jsonb(a)
```